ra/registration-authority.go

```diff
@@ -640,6 +640,21 @@ func (ra *RegistrationAuthorityImpl) checkCertificatesPerNameLimit(names []strin
 	return nil
 }
 
+func (ra *RegistrationAuthorityImpl) checkCertificatesPerNameSetLimit(names []string, limit cmd.RateLimitPolicy) error {
+	hash := core.HashNames(names)
+	count, err := ra.SA.CountValidNameSets(hash)
+	if err != nil {
+		return err
+	}
+	if int(count) > limit.GetThreshold(fmt.Sprintf("%X", hash), 0) { // this should prob be plaintext not a hash
```

Comment:
I think it's okay for the key-based override to be based on a hash. I suspect in practice we're unlikely to do many key-based overrides for this rate limit.